### PR TITLE
doc: Organize OEPs by Type

### PR DESCRIPTION
**Pull request by feanil**
Make it easier to seperate historical decision, best practices and
process documentation.  Seperate them on disk and then put them in
separate sections in the index for easing browsability.

### CODE REVIEW

File: oeps/index.rst [hunk 1]
```diff
@@ -24,12 +24,45 @@ archive of large and broadly relevant choices made for the platform.
...
+   :caption: Best Practices
+
+   best-practices/oep-*
+
+Architectural Decisions
+~~~~~~~~~~~~~~~~~~~~~~~
+
+OEPs that capture architectural decisions that impact the entire Open edX
+Platform. These are decisions that are sufficiently cross-cutting that there
+is no clear code repositary where they should reside.
```
Comment (davidjoy) [inline oeps/index.rst:58]:
```suggestion
is no clear code repository where they should reside.
```

**Review by davidjoy — APPROVED** [review]
One nit spelling error.  Nice organization.

**Comment by idegtiarov** [comment]
Like that change, thank you @feanil.

**Comment by e0d** [comment]
@feanil what blocks us from merging this?

**Pull request merged**